common/canonical_serialization/README.md

```diff
@@ -30,13 +30,13 @@ Unless specified, all numbers are stored in little endian format.
 
 ### Booleans and Integers
 
-|Type	                    |Original data	        |Hex representation	|Serialized format	|
-|---	                    |---	                |---	            |---	            |
-|Boolean	                |True / False	        |0x01 / 0x00	    |[01] / [00]	    |
-|8-bt unsigned integer	    |1	                    |0x01	            |[01]	            |
-|16-bit unsigned integer	|4660	                |0x1234	            |[3412]	            |
-|32-bit unsigned integer	|305419896	            |0x12345678	        |[78563412]	        |
-|64-bit unsigned integer	|1311768467750121216	|0x12345678ABCDEF00	|[00EFCDAB78563412]	|
+|Type                       |Original data          |Hex representation |Serialized format  |
+|---                        |---                    |---                |---                |
+|Boolean                    |True / False           |0x01 / 0x00        |[01] / [00]        |
+|8-bt unsigned integer      |1                      |0x01               |[01]               |
+|16-bit unsigned integer    |4660                   |0x1234             |[3412]             |
+|32-bit unsigned integer    |305419896              |0x12345678         |[78563412]         |
+|64-bit unsigned integer    |1311768467750121216    |0x12345678ABCDEF00 |[00EFCDAB78563412] |
 
 ### Byte Arrays
 
```

Comment:
I think I'd commented on this before, but not sure what the resolution was.  It doesn't seem like it would be canonical if you allow this to be optional.  For example, if there is a byte array in the rawTransaction object, we can't have the client sometimes serializing+signing one format and sometimes another

Reply:
It is still canonical because the specific representation of data is left up to the author of a specification. Specifically in a spec, the author would have to note whether or not the field was a fixed length array or pre-fixed length array. Using your example, a client that does not conform to the spec would be speaking gibberish. I do agree it definitely adds overhead and potential confusion. On the flip side, there exists a few uses of the fixed length (no prefix) byte arrays in testnet. Perhaps the wording can be improved that it is optional when defining the spec but they are not interchangeable once defined.

Reply:
Why have this confusion though?  Why not just always specify it?  You aren't really saving that much by making it optional

Reply:
Agreed. I think for now, we can remove this notion as it doesn't affect the RawTransaction code. I'll file an issue to address this in the validator code and remove it from the rust implementation.